kolibri/core/assets/src/composables/useMinimumKolibriVersion.js

```
@@ -0,0 +1,35 @@
+import { computed } from 'kolibri.lib.vueCompositionApi';
+
+export default function useMinimumKolibriVersion() {
+  const isMinimumKolibriVersion = computed(() => {
+    return function(version, majorVersion = 0, minorVersion = 15, revisionVersion = null) {
+      /*
+      defaults to kolibri 0.15.x
+      If values are not provided for revisionVersion, then any values are allowed.
+      This means that if no value is provided for revisionVersion then
+      alpha and beta versions will be permitted.
+      Just providing 0 as majorVersion and 15 as minorVersion would allow any version
+      greater than or equal to 0.15 (including any alphas or betas)
+      */
+      const v = version.split('.');
```

I know we're not there yet :-) but I'd expect `isMinimumKolibriVersion('1', 0, 15)` and `isMinimumKolibriVersion('1.0', 0, 15)` to return `true`, the same way as `isMinimumKolibriVersion('1.0.0', 0, 15)` does. Would that make sense? Strictly speaking, I'm not actually sure if v1 can be considered the same v1.0 and v1.0.0, it's rather my intuition.

In any case, from a couple of places in this function, it seems that all three parts are needed, so on the line below, maybe we could just throw an error if the version value doesn't satisfy the full format requirement?

```
if (v.length < 3) {
   throw new Error('The full version format (e.g. 0.15.0) is required')
}
```

To ensure that it won't fail silently when returning `false` only because of an unexpected parameter, rather than as a result of comparing versions.

I'd also recommend checking that `version` is defined before trying to split it.

---

I was talking about t his with Richard, if we want to have a complete flexibility on the version naming,  we should use something like https://www.npmjs.com/package/compare-versions 
However, in this case, as we're who are the users of the function and kolibri is going to be a '0.0.0' version always we don't need to add one more dependency. So, yes , you are right for that case... but that case is not going to happen.

---

No need for complete flexibility. That's up to you. I imagine that at least communicating that such a decision has been made would help, with a comment or the parameter check. v1 was just one example, this will currently silently fail for non-v1 values too, like  `isMinimumKolibriVersion.value('0.16', 0, 15, 0)` returns `false` whereas `isMinimumKolibriVersion.value('0.16.0', 0, 15, 0)` returns `true`.

---

the thing is that this is going to be used to check the version provided by kolibri, and kolibri will always provide three digits. I can add a comment in the function but, while this is been done only to be used inside kolibri, there should not be possible error.

---

Everything's possible, Jose, Kolibri is written by humans, not machines (at least not yet) ;-)) Leaving philosophy behind, this is not a blocker when compared to console errors, so it's up to you.

---

yes, you're right. I try to keep it simple (or using the external library would be a better option) but throwing an error as you suggest is harmless if we keep the format as it's supposed and does not add complexity. Done!